codes/typescript/chapter_hashing/array_hash_map.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * File: array_hash_map.ts
+ * Created Time: 2022-12-20
+ * Author: Daniel (better.sunjian@gmail.com)
+ */
+
+/* 键值对 Number -> String */
+class Entry {
+  public key: number;
+  public val: string;
+  constructor(key: number, val: string) {
+    this.key = key;
+    this.val = val;
+  }
+}
+
+/* 数组的初始化和基本操作 */
+class ArrayList {
```

Comment:
Could we use the built-in list `[]` instead?  You don't have to implement the ArrayList in Java. `[]` is the ArrayList in JS and TS.